File: src/main/java/io/beaniejoy/dongnecafe/domain/cafe/dto/menu/MenuOptionResponseDto.java
package io.beaniejoy.dongnecafe.domain.cafe.dto.menu;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.List;
import java.util.UUID;

@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class MenuOptionResponseDto {

    private UUID optionId;

    private String title;

    private List<OptionDetailResponseDto> optionDetailList;
}

File: src/main/java/io/beaniejoy/dongnecafe/domain/cafe/dto/menu/MenuOptionResponseDto.kt
package io.beaniejoy.dongnecafe.domain.cafe.dto.menu

import io.beaniejoy.dongnecafe.domain.cafe.entity.MenuOption

data class MenuOptionResponseDto(
    val id: Long = 0L,
    val title: String? = null,
    val optionDetailList: List<OptionDetailResponseDto> = emptyList()
) {
    companion object {
        fun of(menuOption: MenuOption): MenuOptionResponseDto {
            return MenuOptionResponseDto(
                id = menuOption.id,
                title = menuOption.title,
                optionDetailList = menuOption.optionDetailList.map { OptionDetailResponseDto.of(it) }
            )
        }
    }
}